src/twisted/trial/_asynctest.py

```diff
@@ -205,8 +205,6 @@ def deferRunCleanups(self, ignored, result):
 
         for f in failures:
             result.addError(self, f)
-            if f.check(KeyboardInterrupt):
-                result.stop()
```

Comment:
Even though this condition is never true, is removing the check an improvement or does it takes us even further from the desired behavior?

Reply:
removing unreachable code is an improvement, swapping this for AsyncExitStack and SyncExitStack could be done in a followup?

Reply:
Any reactor that inherits `_SignalReactorMixin` replaces the default `SIGINT` handler that raises `KeyboardInterrupt` with one that doesn't, so in practice cleanup handlers cannot be aborted with CTRL-C. Addressing this issue is going to require more invasive changes and then indeed it doesn't help to keep this code around.